[COLLAB] Design arbiter volume support

phlogistonjohn opened the pull request:
As suggested by @raghavendra-talur we can collaborate on the design doc within this PR.

phlogistonjohn commented:
I just updated this with a big chunk of proposed code for new interfaces that would be a "new allocator" layer. Please read and comment.

phlogistonjohn commented:
I pushed a new section on the flags/tags for devices and nodes. We need to discuss this some more.

humblec commented on docs/design/arbiter.md:
```diff
@@ -0,0 +1,191 @@
+This is a summary of the areas of the code that will need to be modified
+to support arbiter volumes:
+
+* Volume Create
```
may be we can mention the key `arbiter` on volume options in heketi cli or api client can trigger these kind of volumes.

phlogistonjohn commented:
I've rebased and updated the document in regards to recent discussions. I've also removed the do-not-merge flag, as its probably in a good enough state now that a good chunk of it is implemented. Plus, there's nothing stopping us from updating it in new PRs.

obnoxxx commented:
@phlogistonjohn We should squash these to 1 commit and merge it.

obnoxxx commented:
@phlogistonjohn ping. Any opinions on squashing and then merging? :-)

obnoxxx commented:
To elaborate, the commit could be expanded like so:

```
commit 0965b0cc8b31ed0e9b06ce347ac76f84a811858e (HEAD -> pr/1067)
Author:     Michael Adam <obnox@redhat.com>
AuthorDate: Mon Feb 26 15:57:25 2018 +0100

    docs: create an arbiter design doc
    
    This captures the design of the arbiter feature as discussed between
    2018-02-27 and 2018-03-19. The doc largely captures what was added to
    Heketi minus some differences that occurred during development.
    
    Pair-Programmed-With: John Mulligan <jmulligan@redhat.com>
    Pair-Programmed-With: Raghavendra Talur <rtalur@redhat.com>
    
    Signed-off-by: Michael Adam <obnox@redhat.com>
    Signed-off-by: John Mulligan <jmulligan@redhat.com>
    Signed-off-by: Raghavendra Talur <rtalur@redhat.com>
```

But @phlogistonjohn or @raghavendra-talur would be just as fine as author, with the corresponding changes to the pair-programmed lines.

obnoxxx commented:
Alternative:
```
commit 81e17526d52d80067292b784edb26d8199160a42 (HEAD -> pr/1067)
Author:     John Mulligan <jmulligan@redhat.com>
AuthorDate: Mon Feb 26 15:57:25 2018 +0100

    docs: create an arbiter design doc
    
    This captures the design of the arbiter feature as discussed between
    2018-02-27 and 2018-03-19. The doc largely captures what was added to
    Heketi minus some differences that occurred during development.
    
    Pair-Programmed-With: Michael Adam <obnox@redhat.com>
    Pair-Programmed-With: Raghavendra Talur <rtalur@redhat.com>
    
    Signed-off-by: John Mulligan <jmulligan@redhat.com>
    Signed-off-by: Michael Adam <obnox@redhat.com>
    Signed-off-by: Raghavendra Talur <rtalur@redhat.com>
```
